invenio_records_resources/resources/records/resource.py

```diff
@@ -106,6 +108,12 @@ def read(self):
             resource_requestctx.view_args["pid_value"],
             expand=resource_requestctx.args.get("expand", False),
         )
+
+        # emit record view stats event
+        emitter = current_stats.get_event_emitter("record-view")
+        if item is not None and emitter is not None:
+            emitter(current_app, record=item._record, via_api=True)
```

Comment:
I guess this was already discussed, but I would have instead:
1. move the emitting part inside the `read` service method
2. pass to the service an extra kwargs to give context to that call, e.g. `is_api=[True|False]`, with a default value or a mandatory value (to avoid forgetting to pass it).
 
This would allow code duplication when adding this in the Flask `view`.

Reply:
Summary of the brief discussion just now:
The events fetch some request-dependent information (e.g. client IP address) that isn't necessarily available to the service right now. This information would have to be passed in some reliable way.
Further, it might be that the service layer is called from other places that we do not want to count, e.g. as part of a CLI command (like we do in `Invenio-Utilities-TUW` sometimes).
Basically, it was found that the resources & view functions are an okay place, because those are the calls we're interested in counting.

Reply:
please add a comment in the code as well :)

Reply:
> Summary of the brief discussion just now: The events fetch some request-dependent information (e.g. client IP address) that isn't necessarily available to the service right now. This information would have to be passed in some reliable way. 

On this, not sure what is the issue around "reliable way". In any case, see below.

> Further, it might be that the service layer is called from other places that we do not want to count, e.g. as part of a CLI command (like we do in `Invenio-Utilities-TUW` sometimes).

Agree very much on this and, for example, the fact that a service returns a `result_item`, instead of the record itself, is an issue when calling services from services.
It feels like something we are not doing 100% correctly. Let's go ahead with this to avoid blocking this PR further, we can always refactor later on.